src/navigation/SwipeNavigator.js

```diff
@@ -1,43 +1,387 @@
 import { deviceUtils } from '../utils';
+import {
+  ScrollPositionContext,
+  usePagerPosition,
+} from './ScrollPositionContext';
 import Routes from './routesNames';
-import { useAccountSettings, useCoinListEdited } from '@/hooks';
-import { useTheme } from '@/theme';
+import {
+  useAccountAccentColor,
+  useAccountSettings,
+  useCoinListEdited,
+  useDimensions,
+} from '@/hooks';
+import { Box, Columns, Stack } from '@/design-system';
+import { ButtonPressAnimation } from '@/components/animations';
+
+import logger from 'logger';
+import WalletScreen from '@/screens/WalletScreen';
+import RecyclerListViewScrollToTopProvider, {
+  useRecyclerListViewScrollToTopContext,
+} from '@/navigation/RecyclerListViewScrollToTopContext';
+import { discoverOpenSearchFnRef } from '@/screens/discover/components/DiscoverSearchContainer';
+import { InteractionManager, View } from 'react-native';
+import { IS_IOS } from '@/env';
+import SectionListScrollToTopProvider, {
+  useSectionListScrollToTopContext,
+} from './SectionListScrollToTopContext';
+
+const NUMBER_OF_TABS = 3;
+
+const config = {
+  animation: 'spring',
+  config: {
+    stiffness: 200,
+    damping: 500,
+    mass: 3,
+    overshootClamping: true,
+    restDisplacementThreshold: 0.01,
+    restSpeedThreshold: 0.01,
+  },
+};
+const tabConfig = { duration: 250, easing: Easing.elastic(1) };
 
 const Swipe = createMaterialTopTabNavigator();
+const HEADER_HEIGHT = IS_IOS ? 82 : 62;
+
+const TabBar = ({
+  state,
+  descriptors,
+  navigation,
+  position,
+  isTap,
+  setIsTap,
+  lastPress,
+  setLastPress,
+}) => {
+  const { width: deviceWidth } = useDimensions();
+  const reanimatedPosition = useSharedValue(2);
+  const tabWidth = deviceWidth / NUMBER_OF_TABS;
+  const tabPillStartPosition = (tabWidth - 72) / 2;
+
+  const { accentColor } = useAccountAccentColor();
+  const recyclerList = useRecyclerListViewScrollToTopContext();
+  const sectionList = useSectionListScrollToTopContext();
+  // ////////////////////////////////////////////////////
+  // Colors
+
+  const { colors, isDarkMode } = useTheme();
+  const tabStyle = useAnimatedStyle(() => {
+    const pos1 = tabPillStartPosition;
+    const pos2 = tabPillStartPosition + tabWidth;
+    const pos3 = tabPillStartPosition + tabWidth * 2;
+    const translateX = interpolate(
+      reanimatedPosition.value,
+      [0, 1, 2],
+      [pos1, pos2, pos3],
+      Extrapolate.EXTEND
+    );
+    return {
+      transform: [{ translateX }],
+      width: 72,
+    };
+  });
+
+  const animateTabs = (index, shouldAnimate = false) => {
+    if (shouldAnimate) {
+      reanimatedPosition.value = withTiming(index, tabConfig);
+      return;
+    }
+    reanimatedPosition.value = index;
+  };
+
+  useEffect(() => {
+    if (!isTap) {
+      animateTabs(state.index, true);
+    } else {
+      animateTabs(state.index);
+    }
+    setIsTap(false);
+  }, [state.index, isTap]);
+
+  // for when QRScannerScreen is re-added
```

Comment:
bumping to make sure Christian is aware of this

Reply:
ya we agreed modal nav instead of 4th tab is fine for now